### PR TITLE
hotfix(frontend): fixed Validators page where unknown countryCode would crash the whole page

### PR DESCRIPTION
**Pull request by frol**
@shelegdmitriy I have hotfixed the validators page getting crashed when staking pool has unknown `countryCode` ([`BYAA`](https://en.wikipedia.org/wiki/ISO_3166-3#BYAA) is a correct country code identifier, but the JS packages we use do not recognize it, so neither country name nor flag is pulled in)

image

This is suboptimal, and we should have a better fix here, so this branch is here only for reference (I deployed this fix to production as part of the `hotfix/revert-telemetry` branch).

We should consider that staking pool metadata might even have malicious data, so we should have some strategy on how to handle it properly (e.g. check if countries-data has the information about the country and if it does not have it, we should just fall back to "unknown")

### CODE REVIEW

**Comment by shelegdmitriy** [comment]
@frol you are right. The thing with this example that **BYAA** is in _ISO 3166-3_ format but if we pass **BY** (_Former codes_ column) to our js library - we'll see correct country flag. It's just FYI

File: frontend/src/components/utils/CountryFlag.tsx [hunk 1]
```diff
@@ -20,22 +25,53 @@ const DefaultCountryFlag = () => (
...
+    typeof props.country !== undefined &&
+    countries[props.country.toUpperCase()]
+      ? props.country.toUpperCase()
+      : undefined;
+  const countryCode: string | undefined =
+    props.countryCode && typeof props.countryCode !== undefined
+      ? props.countryCode.toUpperCase()
+      : mayBeCountryCode
+      ? mayBeCountryCode
+      : undefined;
```
Comment (frol) [inline frontend/src/components/utils/CountryFlag.tsx:41]: I find this mix of conditions unreadable

```suggestion
  let countryCode: string | undefined = props.countryCode?.toUpperCase();
  if (!(countryCode in countries)) {
    const countryAsCountryCode = props.country?.toUpperCase();
    if (countryAsCountryCode in countries) {
      countryCode = countryAsCountryCode;
    }
  }
```

File: frontend/src/components/utils/CountryFlag.tsx [hunk 2]
```diff
@@ -20,22 +25,53 @@ const DefaultCountryFlag = () => (
...
+  const countryLabel: string | undefined = countryCode
+    ? countries[countryCode].name
+    : props.country
+    ? props.country
+    : undefined;
```
Comment (frol) [inline frontend/src/components/utils/CountryFlag.tsx:46]:
```suggestion
  const countryLabel: string | undefined = countries[countryCode]?.name ?? props.country;
```

File: frontend/src/components/nodes/ValidatorMainRow.tsx [hunk 3]
```diff
@@ -99,17 +98,11 @@ class ValidatorMainRow extends PureComponent<Props> {
...
-                      {countryCode && typeof countryCode !== undefined
-                        ? countries[countryCode.toUpperCase()].name
-                        : country}
-                    </Tooltip>
-                  }
-                >
-                  <CountryFlag countryCode={countryCode} />
-                </OverlayTrigger>
+                <CountryFlag
+                  id={`${countryCode || "country"}_${index}`}
```
Comment (frol) [inline frontend/src/components/nodes/ValidatorMainRow.tsx:102]: Can we drop this `id`?

Comment (shelegdmitriy) [reply frontend/src/components/nodes/ValidatorMainRow.tsx:102]: No we can't because of warning

**Review by shelegdmitriy — APPROVED** [review]

File: frontend/src/components/nodes/ValidatorMainRow.tsx [hunk 4]
```diff
@@ -99,17 +98,11 @@ class ValidatorMainRow extends PureComponent<Props> {
...
-                      {countryCode && typeof countryCode !== undefined
-                        ? countries[countryCode.toUpperCase()].name
-                        : country}
-                    </Tooltip>
-                  }
-                >
-                  <CountryFlag countryCode={countryCode} />
-                </OverlayTrigger>
+                <CountryFlag
+                  id={`country_flag_${account_id}`}
```
Comment (frol) [inline frontend/src/components/nodes/ValidatorMainRow.tsx:102]:
```suggestion
                  id={`country_flag_${accountId}`}
```

**Pull request merged**